packages/notification-center/src/components/notification-center/components/user-preference/ChannelPreference.tsx

```diff
@@ -1,17 +1,45 @@
 import React from 'react';
-import { Text, TextBlock } from './UserPreference';
 import { useNovuThemeProvider } from '../../../../hooks/use-novu-theme-provider.hook';
 import { Switch } from '@mantine/core';
 import { getChannel } from './channels';
 import styled from 'styled-components';
+import { useApi } from '../../../../hooks';
+import { switchStyles, Text, TextBlock } from './styles';
+import { IUserPreferenceSettings } from '../../../../index';
 
-export function ChannelPreference({ type, active }) {
+interface IChannelPreferenceProps {
+  type: string;
+  active: boolean;
+  templateId: string;
+  index: number;
+  setSettings: (IUserPreferenceSettings) => void;
```

Comment:
```suggestion
  setSettings: (settings: IUserPreferenceSettings) => void;
```


The same one goes here :)